supermarq-benchmarks/supermarq/qcvv/irb.py

```diff
@@ -99,71 +332,97 @@ class IRB(BenchmarkingExperiment[Union[IRBResults, RBResults]]):
+            clifford: The clifford operation to convert.
+
         Returns:
-            The single reduced gate
+            A circuit implementing the desired Clifford gate.
+        """
+        circuit = cirq.Circuit(
+            cirq.decompose_clifford_tableau_to_operations(
+                self.qubits, clifford.clifford_tableau  # type: ignore[arg-type]
+            )
+        )
+        return cirq.optimize_for_target_gateset(circuit, gateset=self.clifford_op_gateset)
+
+    def random_clifford(self) -> cirq.CliffordGate:
+        """Returns:
+        A random clifford gate with the correct number of qubits for the current experiment.
         """
-        cur = gate_seq[0]
-        for gate in gate_seq[1:]:
-            cur = cur.merged_with(gate)
-        return cur
+        if self.num_qubits == 1:
+            return random_single_qubit_clifford()
+        return random_two_qubit_clifford()
 
-    @classmethod
-    def _random_single_qubit_clifford(cls) -> cirq.ops.SingleQubitCliffordGate:
-        """Choose a random single qubit clifford gate.
+    def gates_per_clifford(self, samples: int = 500) -> dict[str, float]:
+        """Samples a number of random Clifford operation and calculates the average number of
+        single and two qubit gates used to implement them. Note this depends on the gateset chosen
+        for the experiment.
+
+        Args:
+            samples: Number of samples to use. Defaults to 500.
 
         Returns:
-            The random clifford gate.
+            A dictionary with the average number of one and two qubit gates used.
         """
-        Id = cirq.ops.SingleQubitCliffordGate.I
-        H = cirq.ops.SingleQubitCliffordGate.H
-        S = cirq.ops.SingleQubitCliffordGate.Z_sqrt
-        X = cirq.ops.SingleQubitCliffordGate.X
-        Y = cirq.ops.SingleQubitCliffordGate.Y
-        Z = cirq.ops.SingleQubitCliffordGate.Z
-
-        set_A = [
-            Id,
-            S,
-            H,
-            cls._reduce_clifford_seq([H, S]),
-            cls._reduce_clifford_seq([S, H]),
-            cls._reduce_clifford_seq([H, S, H]),
+        sample = [
+            self._clifford_gate_to_circuit(self.random_clifford())
+            for _ in tqdm(range(samples), desc="Sampling Clifford operations")
         ]
-
-        set_B = [Id, X, Y, Z]
-
-        return cls._reduce_clifford_seq([random.choice(set_A), random.choice(set_B)])
+        return {
+            "single_qubit_gates": np.mean(
+                [
+                    sum(1 for op in circuit.all_operations() if len(op.qubits) == 1)
+                    for circuit in sample
+                ]
+            ).item(),
+            "two_qubit_gates": np.mean(
+                [
+                    sum(1 for op in circuit.all_operations() if len(op.qubits) == 2)
+                    for circuit in sample
+                ]
+            ).item(),
+        }
 
     def _invert_clifford_circuit(self, circuit: cirq.Circuit) -> cirq.Circuit:
```

Comment:
Perhaps rename to something like:
```suggestion
    def _append_inverse_clifford_circuit(self, circuit: cirq.Circuit) -> cirq.Circuit:
```

Reply:
This function isn't actually used any more so I've removed it.